### PR TITLE
rewrite welcome page (fixed histo)

### PR DESCRIPTION
**Pull request by Skylsmoi**

### CODE REVIEW

File: src/server/kiwix-serve.cpp [hunk 1]
```diff
@@ -879,26 +879,46 @@ int main(int argc, char **argv) {
...
+"}"
+".book:hover { background-color: #eee; box-shadow: 2px 2px 5px 0px #ccc}"
+".book__background { background-repeat: no-repeat; background-size: auto; background-position: top right; }"
+".book__title { font-size: 18px; color: #0645ad; }"
+".book__description { padding: 5px 50px 5px 0px; font-size: 15px; }"
+".book__info { line-height: 18px; color: #777; font-weight: bold; font-size: 13px; }"
+"</style>";
+
+  string welcomeBooksHtml = ""
+"<div class='book__list'>";
```
Comment (Skylsmoi) [inline src/server/kiwix-serve.cpp]: I did it to keep some kind of an indentation for the html.
Creating template in a backend language can quickly make code complicated. I do this to keep the html as much readable as I can.
The other bright side is that the html generated when viewing the source in the browser (right click > inspect source) has a right indentation. A minor addition though.

File: src/server/kiwix-serve.cpp [hunk 2]
```diff
@@ -879,26 +879,46 @@ int main(int argc, char **argv) {
...
+    "display: inline-block; vertical-align: bottom; margin: 10px; padding: 15px 20px; width: 300px;"
+    "border: 1px solid #ccc; border-radius: 15px;"
+    "text-align: left; color: #000; font-family: sans-serif; font-size: 13px;"
+"}"
+".book:hover { background-color: #eee; box-shadow: 2px 2px 5px 0px #ccc}"
+".book__background { background-repeat: no-repeat; background-size: auto; background-position: top right; }"
+".book__title { font-size: 18px; color: #0645ad; }"
+".book__description { padding: 5px 50px 5px 0px; font-size: 15px; }"
+".book__info { line-height: 18px; color: #777; font-weight: bold; font-size: 13px; }"
+"</style>";
```
Comment (Skylsmoi) [inline src/server/kiwix-serve.cpp]: Just to keep the html and it's style at the same place, it's easier to code and debugg.
It's not a bad idea to put it in the template file though

Comment (kelson42) [inline src/server/kiwix-serve.cpp]: Yes, please put as much things as possible in the template.

**Comment by kelson42** [comment]
screen shot 2017-06-07 at 20 08 50
screen shot 2017-06-07 at 20 08 40

Few remarks:
* The boxes do not have the same height, this should be fix probably by ussing "ellipsis".
* Using ellipsis also to avoid the text collision with the icon
* It would be good to have a nice rendering with 360×640, reducing a bit the marge/padding should be good enough
* An idea, I tend to think that having the page background and boxes greyed/darked and bright if you are over the box would make a better rendering. Currently it's exactly the opposite.
* Having something (light grey?) in background using css3 pattern (http://lea.verou.me/css3patterns/) would really help to make the whole stuff "warmer".

Otherwise looks good to me.

**Comment by Skylsmoi** [comment]
In the last commit, I have moved the <style> tag into the template, reduced margin/padding of boxes, implemented the ellipsis css for title and description (so now every boxes will have the same size) and added html "title" on boxes title and description to display the full title or description since there is now an ellipsis

**Comment by kelson42** [comment]
@Skylsmoi Looks good to me! Congrats for all the good work!

**Pull request merged**